Source/Internal/IGListPreprocessor.h

```diff
@@ -0,0 +1,34 @@
+//
+//  IGListPreprocessor.h
+//  IGListKit
+//
+//  Created by Adlai Holler on 2/20/17.
+//  Copyright © 2017 Instagram. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import <IGListKit/IGListMacros.h>
+#import <IGListKit/IGListPreprocessingContext.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class IGListSectionController, IGListSectionMap;
+
+IGLK_SUBCLASSING_RESTRICTED
+@interface IGListPreprocessor : NSObject
+
+- (instancetype)initWithSectionMap:(IGListSectionMap *)sectionMap;
+
+/**
+ * Must be called from the main thread.
+ */
+- (void)schedulePreprocessingWithContainerSize:(CGSize)containerSize completion:(dispatch_block_t)completionBlock;
```

Comment:
oh yes, saw the original issue and read through it. just asking the question if there was another way to get this info considering the sizes may change.

if sizes do change, is a developer expected to kick off the calculations again? how is that handled in this system? does there need to be a way to cancel what may be long running background processes that are no longer valid?